src/miner.cpp

```diff
@@ -858,135 +902,148 @@ bool SignStakeBlock(CBlock &block, CKey &key, vector<const CWalletTx*> &StakeInp
-    if(OutOfSyncByAge())
-    {
-        LogPrintf("AddNeuralContractOrVote: Out Of Sync");
+    if (OutOfSyncByAge()) {
+        LogPrintf("AddNeuralContractOrVote: Out of sync.");
         return;
     }
 
-    if(!IsNeuralNodeParticipant(bb.GRCAddress, blocknew.nTime))
-    {
-        LogPrintf("AddNeuralContractOrVote: Not Participating");
+    std::string quorum_address = DefaultWalletAddress();
+
+    if (!IsNeuralNodeParticipant(quorum_address, blocknew.nTime)) {
+        LogPrintf("AddNeuralContractOrVote: Not participating.");
         return;
     }
 
-    if(blocknew.nVersion >= 9)
-    {
-        // break away from block timing
-        if (fDebug) LogPrintf("AddNeuralContractOrVote: Updating Neural Supermajority (v9 M) height %d",nBestHeight);
-        ComputeNeuralNetworkSupermajorityHashes();
+    if (fDebug) {
+        LogPrintf("AddNeuralContractOrVote: Updating neural supermajority...");
     }
 
-    if(!NeedASuperblock())
-    {
-        LogPrintf("AddNeuralContractOrVote: not Needed");
+    ComputeNeuralNetworkSupermajorityHashes();
+
+    if (!NeedASuperblock()) {
+        LogPrintf("AddNeuralContractOrVote: Not needed.");
         return;
     }
 
     int pending_height = RoundFromString(ReadCache(Section::NEURALSECURITY, "pending").value, 0);
 
-    if (pending_height>=(pindexBest->nHeight-200))
-    {
-        LogPrintf("AddNeuralContractOrVote: already Pending");
+    if (pending_height >= (pindexBest->nHeight - 200)) {
+        LogPrintf("AddNeuralContractOrVote: Already pending.");
         return;
     }
 
     double popularity = 0;
     std::string consensus_hash = GetNeuralNetworkSupermajorityHash(popularity);
 
-    /* Retrive the neural Contract */
-    const std::string& sb_contract = NN::GetInstance()->GetNeuralContract();
-    const std::string& sb_hash = GetQuorumHash(sb_contract);
-
-    if(!sb_contract.empty())
-    {
+    // Add our Neural Vote
+    //
+    // GetSuperblockHash() returns an invalid QuorumHash when the node has not
+    // yet received enough scraper data to resolve a convergence locally so it
+    // cannot vote for a superblock.
+    //
+    blocknew.m_claim.m_quorum_hash = NN::GetInstance()->GetSuperblockHash();
 
-        /* To save network bandwidth, start posting the neural hashes in the
-           CurrentNeuralHash field, so that out of sync neural network nodes can
-           request neural data from those that are already synced and agree with the
-           supermajority over the last 24 hrs
-           Note: CurrentNeuralHash is not actually used for sb validity
-        */
-        bb.CurrentNeuralHash = sb_hash;
+    if (!blocknew.m_claim.m_quorum_hash.Valid()) {
+        LogPrintf("AddNeuralContractOrVote: Local contract empty.");
+        return;
+    }
 
-        /* Add our Neural Vote */
-        bb.NeuralHash = sb_hash;
-        LogPrintf("AddNeuralContractOrVote: Added our Neural Vote %s",sb_hash);
+    blocknew.m_claim.m_quorum_address = std::move(quorum_address);
```

Comment:
Are we conditioning doing the quorum on the move to v11? It appears that condition doesn't exist yet? Maybe I missed it.

Reply:
Not yet--we haven't hooked up the new `Superblock` class to the block acceptance/validation pipeline yet. I tried to match the behavior here as closely to the old code as possible. We can add the check in a subsequent PR that focus on the transition from the quorum.